internal/pkg/agent/application/dispatcher/dispatcher_test.go

```diff
@@ -231,41 +231,38 @@ func TestActionDispatcher(t *testing.T) {
 	})
 
 	t.Run("Cancel queued action", func(t *testing.T) {
-		def := &mockHandler{}
-		calledCh := make(chan bool)
-		call := def.On("Handle", mock.Anything, mock.Anything, mock.Anything).Return(nil).Once()
-		call.RunFn = func(_ mock.Arguments) {
-			calledCh <- true
-		}
-
 		queue := &mockQueue{}
 		queue.On("Save").Return(nil).Once()
 		queue.On("DequeueActions").Return([]fleetapi.ScheduledAction{}).Once()
 
-		d, err := New(nil, t.TempDir(), def, queue)
-		require.NoError(t, err)
-		err = d.Register(&mockAction{}, def)
-		require.NoError(t, err)
-
 		action := &mockAction{}
 		action.On("Type").Return(fleetapi.ActionTypeCancel)
 		action.On("ID").Return("id")
 
 		dispatchCtx, cancelFn := context.WithCancel(context.Background())
 		defer cancelFn()
-		go d.Dispatch(dispatchCtx, detailsSetter, ack, action)
+
+		def := &mockHandler{}
+		def.On("Handle", dispatchCtx, action, ack).Return(nil).Once()
+
+		d, err := New(nil, t.TempDir(), def, queue)
+		require.NoError(t, err)
+
+		dispatchCompleted := make(chan struct{})
+		go func() {
+			d.Dispatch(dispatchCtx, detailsSetter, ack, action)
+			dispatchCompleted <- struct{}{}
+		}()
+
 		select {
 		case err := <-d.Errors():
 			t.Fatalf("Unexpected error: %v", err)
-		case <-calledCh:
-			// Handle was called, expected
-		case <-time.After(1 * time.Second):
-			t.Fatal("mock Handle never called")
+		case <-dispatchCompleted:
+			// OK, expected to complete the dispatch without blocking on the errors channel
```

Comment:
If this never happens and nothing ever comes over `d.Errors()` doesn't this mean this test is now stuck?

I believe the `case <-time.After(1 * time.Second)` was added to fail in this case. Possible this should be kept, but maybe with a longer time? Even if its bumped to 10 seconds, just to prevent it from deadlocking here.

Reply:
The test timeout should take care of this scenario.

I believe the current test timeout is the [default](https://pkg.go.dev/cmd/go#hdr-Testing_flags) 10 minutes, which is a bit lengthy for actual unit tests.